### ISSUE TITLE
Add a whitelist for downloading files from non-SSL web sites

### ISSUE THREAD

**Issue by jeremytieman** [brave/brave-browser#36079]
## Description 
When attempting to download from non-SSL protected sites, the downloader by default refuses to download the file until you confirm. I run a website on my home network that I frequently download files from. This means I now have to click multiple times, rather than once, every time I want to download a file. I would like the ability to whitelist the address so I get the normal download experience.

## Steps to Reproduce

   1. Go to a non-SSL protected site.
   2. Download a file.
   3. The download does not occur and you must click on my downloads to confirm the download.

## Expected result:
The file downloads.

## Reproduces how often: 
Always

## Brave version (brave://version info)
Brave 1.64.69 Chromium 122.0.6261.29 (Official Build) nightly (64-bit)
Revision 3992d5550b2a00310f063f95c597bb87c31b4d47
OS Windows 10 Version 22H2 (Build 19045.3693)

**Comment by Brave-Matt** [comment]
+1 on Community:
https://community.brave.com/t/insecure-download-blocked/542748/4

**Comment by jussihi** [comment]
Please add this feature. This used to be possible with flag `--disable-features=InsecureDownloadWarnings` when starting brave. But it is not working anymore.

Why do you keep breaking things with updates like this?

**Comment by JoRoderick** [comment]
We don't need a nanny. Please stop it with this nonsense. I've been online for 30 years. I don't need a nanny. I am considering going back to a much older version of Brave that ... well ... just worked and wasn't twisting my arm everytime I use it.

**Comment by opal1974** [comment]
This breaks my ability to download backups from my NAS - even after selecting 'keep file' downloading won't resume.

**Comment by jussihi** [comment]
@fmarier is there any updates on this?

My robotic process automation uses BRAVE BROWSER for its automation tasks and my automations have been halted for 3 weeks now. Is there update coming regarding this issue or should I switch my automation tasks over to Chromium/Edge ?

I initially chose Brave because initially it didn't have this sort of bullshit nonsense "features" that make use of the browser harder (or at least they could be turned off, see my comment above about `--disable-features=InsecureDownloadWarnings`). This is very very very frustrating.

**Comment by fmarier** [comment]
Is everybody running into this issue wanting to ignore this check for local IP addresses? (e.g. `192.168.0.0/16`)

**Comment by jussihi** [comment]
@fmarier at least me, but in the referred Brave community site (https://community.brave.com/t/insecure-download-blocked/542748) someone is having trouble downloading from this site: http://mileswmathis.com/updates.html

**Comment by jeremytieman** [comment]
@fmarier I believe the issue is due to lack of SSL connections (which is much less common on home networks such as `192.168.0.0/16` and `10.0.0.0/8`). Chrome used to have an option to be able to launch without this check, but what research I did suggests it was removed upstream and we can now see the issue in Brave.

**Comment by Vidanox** [comment]
I've been using Brave for 7 years now and this great "insecure download blocked" "feature" will make me switch if not resolved, tried every solution out there with no result. Stop forcing garbage that just hinders user experience.

![image](https://github.com/brave/brave-browser/assets/20267225/f925d9ce-8fed-403a-84ec-2cc6543116f6)

**Comment by ccampbell** [comment]
This has been happening to me. If I am trying to download a file from a server I am running on localhost, it should be allowed. Period. 

For me it is even worse because for some reason the button to “Keep” when the warning pops up doesn’t do anything. I can click it 100 times and it never actually lets the download occur. That used to work, but stopped working in a newer version of Brave. Even when going to the download page and clicking “Keep” then “Keep Anyway”, nothing happens. Only the “Discard” button actually does anything.

I already have begun migrating away from Brave. This kind of stuff is completely unacceptable. I hope the “protection” you are providing to people who don’t know any better is worth the aggravation and customer turnover you’ll experience from people who do.

**Comment by Sembiance** [comment]
> I already have begun migrating away from Brave.

I agree with everything you said, but I do want to point out that this flag appears to be a 'chrome' thing, not specific to brave. Yes, brave hasn't improved the experience at all, in terms of providing an easy toggle off for this 'feature' or whatever, but if you try and migrate to another chrome based browser you'll likely experience something similar right away, or at least once they update their chrome to these more annoying recent versions. Firefox is still a thing though.

**Comment by pmorange** [comment]
Exactly the same problem as @ccampbell : Trying to download a file from my NAS on LAN and I can't do that because the "keep" button isn't doing anything...
Very bad user experience.

**Comment by fmarier** [comment]
@pmorange Presumably your NAS is using an IP address like `192.168.x.x` or `10.x.x.x`?

**Comment by pmorange** [comment]
> @pmorange Presumably your NAS is using an IP address like `192.168.x.x` or `10.x.x.x`?

Indeed.
I had to use the workaround of the #unsafely-treat-insecure-origin-as-secure flag in order to be able to download files, but as I said a user shouldn't have to search github and apply workarounds himself just to download a file from his LAN :-)

**Comment by fmarier** [comment]
> as I said a user shouldn't have to search github and apply workarounds himself just to download a file from his LAN

Agreed. The trick is to figure out how to enable that use case while still protecting against insecure file downloads in non-home networks (e.g. the proverbial airport/coffee shop network). I'm going to think about it, but it does look like Chromium's approach (which Brave inherits) is too strict.

**Comment by jussihi** [comment]
@fmarier why is this still not fixed?

**Comment by sagb** [comment]
Brave's blocking is a real headache for old servers' IPMI/iLO. They either can't handle modern SSL or mix HTTP/HTTPS, serving a Java viewer over HTTP even when logged in via HTTPS. Brave treats this like it's stopping a bunch of robbers from raiding grandma's bank account.

**Comment by jamincollins** [comment]
How about adding an option to simply turn it off?

I've tried:
* Settings -> Privacy & Security -> Security -> Safe Browsing - No protection
* brave://flags/#brave-override-download-danger-level - Enabled

Nothing seems to turn this off.

I get that you want a better set of defaults eventually.  However, many of us know what we are doing and just want to turn this off so it stops getting in our way.

**Comment by michael-hart-github** [comment]
@ccampbell @mora-phi 
Small workaround for issues with the `keep` button not working

`Settings` -> `Downloads` -> (relevant file) `three dots` -> `Download insecure file`

![Image](https://github.com/user-attachments/assets/4876c725-b566-42f6-b117-08f19e911034)

The `keep` button may be wonky, but at least you a means to download the file(s). 😄 

Personally, I've been dealing with this whenever downloading music I paid for on [7digital](https://us.7digital.com/). They package music files in .zip, which are flagged by Brave. Very annoying. I can't imagine the website is thrilled that their products are marked "insecure", either. 😬

**Comment by fmarier** [comment]
I do agree we should do something about the kinds of connections where TLS is not an option or where it could be considered secure enough (e.g. your own local network).

We are **not** however planning to relax these upstream Chromium checks to remove the warning from sites like 7digital who serve downloads over insecure HTTP. That's a bug on their end that should be reported to them.

**Comment by michael-hart-github** [comment]
@fmarier Sounds good. I wasn't suggesting outside networks should be, I was just showing an example of when I've run into it. 😀 

I've filed a support ticket with 7digital to make them aware of the issue. I figured it was probably some sort of configuration issue on their part. I did take a look at their SSL certs and didn't see anything glaring. 🤷‍♂

**Comment by TheGreeMug** [comment]
+1
This has to be fixed.
Cannot download files from NAS or JellyFin.

HNY

**Comment by mikulik86** [comment]
WTF have I to manually unblock download from https://signal.org/download/???
This is EXTREMELY annoying. Just give us a option to disable the blocking!!

**Comment by mikulik86** [comment]
> Furthermore, it works fine for me

Now works for me too. Very strange behavior!!

**Comment by jrm523** [comment]
Please resolve asap! This is very frustrating. Please allow me to disable safe browsing completely and disable anything that will automatically block my downloads.  I am intelligent enough to know if the websites i visit and files i download are malicious.

**Comment by jussihi** [comment]
@fmarier it has been over one year. Fixing this cannot be this hard. You keep adding fancy useless crypto **** to your borwser yet cannot add an option to whitelist LAN HTTP downloads in 12 months of time. What the ****?

**Comment by jussihi** [comment]
FYI, I have migrated all our systems' browsers to LibreWolf.

This being still a problem is unacceptable. 15 months.

**Comment by krvstn** [comment]
This function is idiotic.

**Comment by danshelp** [comment]
Judging by the age of this discussion, and the fact Brave still won't let me whitelist downloads over my LAN, I take it this is a "feature" we'll be forced to live with if we continue with Brave. Fortunately, other browsers are available.

**Comment by AntiSol** [comment]
This security theatre nonsense needs to go away.

> We are not however planning to relax these upstream Chromium checks to remove the warning from sites like 7digital who serve downloads over insecure HTTP. That's a bug on their end that should be reported to them.

Why, exactly, is it a "bug" to serve files over http?

Let's assume for the sake of argument that serving up a file using a protocol that has been standardised and supported by web browsers for 30+ years is somehow "a bug" now, even though it wasn't a year ago: 

How, exactly, do I contact the deceased owner of a 20 year old site which is only being served via http to report this "bug"?

Note that "not encrypted" is **NOT** the same thing as "insecure", in much the same way as a file being downloaded over an encrypted link does not automatically make it "Secure". Which makes that the wording on the message about "insecure downloads" alarmist and technically incorrect.

This "feature" **only** hinders my experience of brave. And it happens surprisingly frequently. Often I tell brave to download many files, and because the desktop version doesn't actually show any dialog to prompt the user to actually download the file that the user specifically clicked on, indicating a specific desire to download the file, I don't realise until I actually go looking for the download that ha, lol, no: brave decided for me that I'm too stupid to know what I want and that I didn't actually want to download the 50 files that I intentionally clicked "download" on. And so now I have to go into the download manager and manually download all 50 again. Let's just hope that the download wasn't associated with a session that had a timeout. 

If brave wants to act as a nanny and assume that its users are imbeciles who don't know what they're doing, then fine, but you need to provide an option to disable this idiotic feature for people who do actually know what they're doing, and who have been downloading things for 30 years with a single click.

If you're really sure that adding this extra friction to the lives of users is completely neccesary "for teh securitah", then this "feature" needs to at least be optional, so that people who actually know what they're doing aren't penalised.

**Comment by AntiSol** [comment]
Aaaaaaand I just had the same thing again, where I didn't realise it hadn't actually downloaded until I went to open the file.

I particularly love the ridiculousness of showing a save dialog to allow the user to choose where to save a file that you're about to refuse to download. That's not at all an antipattern that defies 30+ years of UI expectation where a user a) sees a save dialog, b) chooses where to save the file and what to name it, and c) presses save, then d) expecting that pressing save _actually saves the goddamn file like the user specifically told the software to_, as opposed to **silently** refusing to do the thing you were just specifically told to do.
